File: solutions/python/isbn-verifier/1/isbn_verifier.py
def verify(isbn):
    sum = 0
    di = 0
    if len(isbn) == 0:
        return False
    isbn = isbn.lower()
    for ni in range(len(isbn)):
        n = isbn[ni]
        if n in "0123456789":
            sum += (10-di)*int(n)
            di += 1
            #print(ni,di,int(n),sum)
        elif n == "x":
            sum += (10-di)*10
        elif n != "-":
            return False
    if sum % 11 == 0:
        return True
    else:
        return False

print (verify('3-598-21508-8'))

File: solutions/fortran/isbn-verifier/1/isbn_verifier.f90
module isbn_verifier
  implicit none

contains

  function isValid(isbn) result(valid)
    character(*), intent(in) :: isbn
    logical :: valid
    character(12) :: vchr = "-0123456789X"
    character(20) :: nums
    integer :: i, sum

    !print *, "isbn=", isbn
    valid = .false.
    
    sum = 0
    nums = ""
    if (index(isbn, "X") > 0 .and. index(isbn, "X") /= len(isbn)) return 
    do i = 1, len(isbn)
        if (index(vchr, isbn(i:i)) == 0) return
        if (isbn(i:i) /= '-') then
                nums = trim(nums) // isbn(i:i) 
        end if
    end do    
    !print *, "nums=", nums
    if (len(trim(nums)) /= 10) return
    do i = 1, len(trim(nums))
                sum = sum + ((11 - i) * (index(vchr, nums(i:i)) - 2))
                !print *, "nums(i,i)=", nums(i:i), " n=", (index(vchr, nums(i:i)) - 2)
    end do

    if (mod(sum, 11) == 0) valid = .true.

  end function isValid

end module isbn_verifier
